payload/payload.cc

```diff
@@ -10,68 +10,73 @@ namespace sorbet::payload {
 
 void createInitialGlobalState(unique_ptr<core::GlobalState> &gs, const realmain::options::Options &options,
                               const unique_ptr<const OwnedKeyValueStore> &kvstore) {
-    if (kvstore) {
-        ENFORCE(!options.noStdlib, "--no-stdlib and --cache-dir are incompatible");
-        auto maybeGsBytes = kvstore->read(core::serialize::Serializer::GLOBAL_STATE_KEY);
-        if (maybeGsBytes.data != nullptr) {
-            Timer timeit(gs->tracer(), "read_global_state.kvstore");
-            core::serialize::Serializer::loadGlobalState(*gs, maybeGsBytes.data);
-            for (unsigned int i = 1; i < gs->filesUsed(); i++) {
-                core::FileRef fref(i);
-                if (fref.dataAllowingUnsafe(*gs).sourceType == core::File::Type::Normal) {
-                    gs = core::GlobalState::markFileAsTombStone(move(gs), fref);
-                }
-            }
-            return;
-        }
-    }
     if (options.noStdlib) {
```

Comment:
minor, optional: should we early return here and ENFORCE that `kvstore` is `nullptr`, so that we don't have to change the indent level of the rest of the block and the diff is smaller?

Reply:
Yep, that would be a nice change.